cumulusci/utils.py

```diff
@@ -382,37 +381,118 @@ def doc_task(task_name, task_config, project_config=None, org_config=None):
+        doc.extend(task_option_doc)
+
+    task_help_doc = create_task_help_doc(task_config)
+    if task_help_doc:
+        doc.append(
+            f"Help for task {task_name}\n------------------------------------------\n"
+        )
+        doc.extend(task_help_doc)
 
     return "\n".join(doc)
 
 
+def get_command_syntax(task_name):
+    """Return an example command syntax string in .rst format"""
+    return f"``$ cci task run {task_name}``\n\n"
+
+
+def get_task_option_info(task_config, task_class):
+    """Gets the the following info for each option in the task
+    usage: example usage statement (i.e. -o name VALUE)
+    required: True/False
+    default: If a default value is present
+    description: Description string provided on the task option
+    option_type: A type string provided on the task option
+
+    Returns list of option dicts with required at the front of the map
+    """
+    required_options = []
+    optional_options = []
+    defaults = task_config.options or {}
+
+    for name, option in list(task_class.task_options.items()):
+        usage = get_option_usage_string(name, option)
+        required = True if option.get("required") else False
+        default = defaults.get(name)
+        description = option.get("description")
+        option_type = option.get("type")
+
+        info = {
+            "usage": usage,
+            "name": name,
+            "required": required,
+            "default": default,
+            "description": description,
+            "option_type": option_type,
+        }
+        if required:
+            required_options.append(info)
+        else:
+            optional_options.append(info)
+
+    return [*required_options, *optional_options]
+
+
+def get_option_usage_string(name, option):
+    """Returns a usage string if one exists
+    else creates a usage string in the form of:
+        -o option_name OPTIONNAME
+    """
+    usage_str = option.get("usage")
+    if not usage_str:
+        usage_str = f"-o {name} {str.upper(name.replace('_',''))}"
+    return usage_str
+
+
+def create_task_options_doc(task_options):
+    """Generate the 'Options' section for a given tasks documentation"""
+    doc = []
+    for option in task_options:
+        usage_str = option.get("usage")
+        if usage_str:
+            doc.append(f"\n``[{usage_str}]``")
+
+        if option.get("required"):
+            doc.append(f"\t *Required*")
+        else:
+            doc.append(f"\t *Optional*")
+
+        description = option.get("description")
+        if description:
+            doc.append(f"\n\t {description}")
+
+        default = option.get("default")
+        if default:
+            doc.append(f"\n\t Default: {default}")
+
+        option_type = option.get("option_type")
+        if option_type:
+            doc.append(f"\n\t Type: {option_type}")
+
+    return doc
+
+
+def create_task_help_doc(task_config):
+    """Create documentation for the 'help' section of a task doc"""
+    return []
```

Comment:
?

Reply:
My intention here was to mimic what is currently in the help section of sfdx command documentation. I'm thinking that if I include task_docs I can take this part out for now.